js/components/portal-dashboard/answers/image-answer.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useState }  from "react";
+// TODO: replace with new model
+import ImageAnswerModal from "../../report/image-answer-modal";
+import { MagnifyIcon } from "./magnify-icon";
+import useResizeObserver from "@react-hook/resize-observer";
+
+import css from "../../../../css/portal-dashboard/answers/image-answer.less";
+
+interface IProps {
+  answer: Map<any, any>;
+  staticSize?: boolean;
+}
+
+const kStaticHeight = 250;
+const kStaticWidth = 250;
+
+export const ImageAnswer: React.FC<IProps> = (props) => {
+  const [modalOpen, setModalOpen] = useState(false);
+
+  const handleShowModal = (show: boolean) => () => {
+    setModalOpen(show);
+  };
+
+  const { answer, staticSize } = props;
+  const imgAnswer = answer.get("answer");
+
+  // cf. https://www.npmjs.com/package/@react-hook/resize-observer
+  const useSize = (target: any) => {
+    const [size, setSize] = React.useState();
+
+    React.useLayoutEffect(() => {
+      setSize(target.current.getBoundingClientRect());
+    }, [target]);
+
+    useResizeObserver(target, (entry: any) => setSize(entry.contentRect));
+    return size;
+  };
+
+  const divTarget = React.useRef(null);
+  const divSize: any = useSize(divTarget);
```

Comment:
Can't we do better than `any` here?

Reply:
Hmmm.... perhaps a `ClientRect`, but then it can also be undefined.  Which if the undefined is added to the type it results in the "property doesn't exist on type never" warning.

Reply:
Actually `DOMRectReadOnly` might be the right choice.  But I still have the "type never" warning to get around.

Reply:
`const divSize: DOMRect | DOMRectReadOnly = useSize(divTarget);` works (the types that we can get from `target.current.getBoundingClientRect()` and `entry.contentRect`) once I also specified an initial state inside of `useSize`.